dts nucleo_f401re flash partition

jamike opened the pull request:
Add partition to be able to build application in slot 0
    with  a file called nucleo_f401re.overlay :
    / {
        chosen {
                zephyr,code-partition = &slot0_partition;
        };
    };
    
    present in application directory tree.

jamike commented:
@rmbolivar  
here is the dts partition.

mbolivar commented on dts/arm/nucleo_f401re.dts:
```diff
@@ -27,3 +27,31 @@
+		slot1_partition: partition@40000 {
+			 label = "image-1";
+			 reg = <0x00040000 0x00020000>;
+		 };
```
Looks like you've got some whitespace issues in this file.

galak commented on dts/arm/nucleo_f401re.dts:
```diff
@@ -27,3 +27,31 @@
 };
+
+&flash0 {
+	partitions {
```
Can you add this comment:

+                               /*
+                                * If chosen's zephyr,code-partition
+                                * is unset, the image will be linked
+                                * into the entire flash device.  If
+                                * it points to an individual
+                                * partition, the code will be linked
+                                * to, and restricted to that
+                                * partition.
+                                */

galak reviewed (changes requested):
add comment about things

mbolivar commented on dts/arm/nucleo_f401re.dts:
```diff
@@ -27,3 +27,31 @@
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
```
This assumes that the processor is not booting from SRAM. Perhaps this should start at 0x08000000, and the other partitions be changed to match, so it will always work.

mbolivar replied:
After discussing with Andy, this is OK -- partitions are defined as an offset / size, not address / size, in each reg.

But there is another bug elsewhere that needs to be addressed. This commit is OK though.

mbolivar commented:
I also think it'd be better to have an application state partition to keep it consistent with the NXP partition table added by @d3zd3z.

jamike commented:
recheck

galak commented:
@mbolivar are you good with this now?

mbolivar commented:
> @mbolivar are you good with this now?

I don't see any changes since my last comment; am I missing something? It seems like my comment about an application state partition still remains.

jamike committed:
dts: nucleo_f401re: add partition support for bootloader

Define the built partition with file nucleo_f401re.overlay
in application directory tree.

Signed-off-by: Michel Jaouen <michel.jaouen@st.com>

agross-oss reviewed (approved):
last commit looks fine to me.  Some odd error.  needs to be rerun

mbolivar commented:
Yup, LGTM.

jamike commented:
let's merge it

galak commented:
If you can do one more rebase, will merge

galak committed:
Merge branch 'arm' into dts_nucleo_f401re_flash_partion
